tracer/src/Datadog.Trace/AppSec/ObjectExtractor.cs

```diff
@@ -142,14 +142,15 @@ private static object ExtractType(Type itemType, object value, int depth, HashSe
             {
                 var dictKey = keyProp.GetValue(item)?.ToString();
                 var dictValue = valueProp.GetValue(item);
-                if (dictValue is null || IsOurKindOfPrimitive(dictValue.GetType()))
+
+                if (dictValue is null)
                 {
-                    items.Add(dictKey, dictValue?.ToString());
+                    items.Add(dictKey, null);
                 }
                 else
                 {
-                    var nestedDict = ExtractProperties(dictValue, depth, visited);
-                    items.Add(dictKey, nestedDict);
+                    var extractedvalue = ExtractType(dictValue.GetType(), dictValue, depth + 1, visited);
```

Comment:
ooc, why the +1 here?

Reply:
The missing + 1 was a bug, I think. We should increment the depth at each recursive call, so we track how deep the object graph gets correctly.

Reply:
unit test added in 2ea687624